src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -2247,7 +2275,20 @@ impl<'a> TypeChecker<'a> {
                         let box (scalar, _) = self.resolve(args[0]).await?;
 
                         let expr = scalar.as_expr()?;
-                        check_number::<_, i64>(span, &self.func_ctx, &expr, &BUILTIN_FUNCTIONS)?
+                        match expr {
```

Comment:
It looks like this code is repeated many times that it can be wrapped into a function to reduce the duplicate codes.

Reply:
The separation allows for more accurate differentiation of error information